akka-actor/src/main/scala/akka/actor/Props.scala

```diff
@@ -130,6 +131,21 @@ object Props {
   def create[T <: Actor](actorClass: Class[T], creator: Creator[T]): Props = {
     apply(defaultDeploy, classOf[CreatorConsumer], actorClass :: creator :: Nil)
   }
+
+  private final def isNotClosingOverEnclosingClass(clazz: Class[_]): Boolean = {
+    def findMatchingConstructor(argTypes: Array[Class[_]]): Boolean =
+      try { val c = clazz.getConstructor(argTypes: _*); true }
+      catch { case ex: NoSuchMethodException ⇒ false }
+    def findMatchingDeclaredConstructor(argTypes: Array[Class[_]]): Boolean =
+      try { val c = clazz.getDeclaredConstructor(argTypes: _*); true }
+      catch { case ex: NoSuchMethodException ⇒ false }
+
+    //noinspection EmptyCheck
+    def defaultEmptyConstructor = clazz.getConstructors.length == 0 && findMatchingDeclaredConstructor(Array.empty)
+    def matchingConstructor = clazz.getDeclaredConstructors.exists { c ⇒ findMatchingConstructor(c.getParameterTypes) }
```

Comment:
My point was that this check can be replace with `clazz.getConstructors.length > 0` AFAICS.